### PR TITLE
feat(tabs): new light/dark mode colors

### PR DESCRIPTION
**Pull request by ze-flo**
## Description

Adds new light/dark mode colors to `Tabs`
## Detail
- Adds new light/dark mode colors to `Tabs`
- Refactors using transient props

## Checklist

- [ ] :ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [ ] ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- [ ] ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [x] :wheelchair: tested for WCAG 2.1 AA accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

File: packages/tabs/src/styled/StyledTab.ts [hunk 1]
```diff
@@ -8,31 +8,36 @@
...
+  $isVertical
 }: IStyledTabProps & ThemeProps<DefaultTheme>) => {
-  const borderColor = isSelected ? 'currentcolor' : 'transparent';
-  const selectedColor = getColorV8('primaryHue', 600, theme);
+  const borderColor = $isSelected ? 'currentcolor' : 'transparent';
+  const borderBlockEndColor = $isVertical ? undefined : borderColor;
+  const borderInlineColor = $isVertical ? borderColor : undefined;
+
+  const selectedColor = getColor({ theme, variable: 'foreground.primary' });
+  const foregroundColor = $isSelected ? selectedColor : 'inherit';
```
Comment (ze-flo) [inline packages/tabs/src/styled/StyledTab.ts:34]: @steue @lucijanblagonic The default foreground color was originally set to inherit the parent container's. In most cases (including the Storybook stories) that would be `foreground.default`, which has the benefit of showing a clear distinction between default and disabled tabs.

Screenshot 2024-06-20 at 4 33 14 PM
Screenshot 2024-06-20 at 4 33 02 PM

Should we keep it as such or use `foreground.subtle` as shown in Figma instead?

CC: @jzempel

Comment (geotrev) [reply packages/tabs/src/styled/StyledTab.ts:34]: Inheriting feels OK here as long as the intent is preserved by the variable (e.g., disabled vs selected vs. default (inherited)). Hopefully I'm understanding this correctly. haha

Comment (lucijanblagonic) [reply packages/tabs/src/styled/StyledTab.ts:34]: @ze-flo Yes, the `foreground.default` is the preferred color here. We corrected this in design, and we left a note for making the change in the inventory.

**Comment by lucijanblagonic** [comment]
There seems to be a flash of black, when the selection indicator is changed. This is only visible for users who rely on keyboard to change the tab. I am assuming that the `foreground.primary` and `hover` mask the black for pointer users.

tabs-flash

image

**Review by jzempel — APPROVED** [review]

**Comment by jzempel** [comment]
> There seems to be a flash of black, when the selection indicator is changed. This is only visible for users who rely on keyboard to change the tab. I am assuming that the `foreground.primary` and `hover` mask the black for pointer users. In dark mode, the flash is in white, `foreground.default` probably.

The only reason we're not seeing this in the [current](https://zendeskgarden.github.io/react-components/?path=/story/packages-tabs-tabs--uncontrolled) is due to the fact that the contrast between the inherited foreground and primary blue is not as high. We might need to question whether the border `transition` ease is serving us well here :shrug: (or could a very slight `transition-delay` fix things?).

**Review by lucijanblagonic — APPROVED** [review]
Everything else looks OK.

**Comment by ze-flo** [comment]
> > There seems to be a flash of black, when the selection indicator is changed. This is only visible for users who rely on keyboard to change the tab. I am assuming that the `foreground.primary` and `hover` mask the black for pointer users. In dark mode, the flash is in white, `foreground.default` probably.
> 
> The only reason we're not seeing this in the [current](https://zendeskgarden.github.io/react-components/?path=/story/packages-tabs-tabs--uncontrolled) is due to the fact that the contrast between the inherited foreground and primary blue is not as high. We might need to question whether the border `transition` ease is serving us well here 🤷 (or could a very slight `transition-delay` fix things?).

The last [commit](https://github.com/zendeskgarden/react-components/pull/1843/commits/093f9c6f5fb8836c5506a85f7bcb20dd698ab4d5) removes the color flash when using the keyboard. ✅

**Review by jzempel — APPROVED** [review]
nice 🙌

**Review by geotrev — APPROVED** [review]

**Review by steue — APPROVED** [review]
looks good! ty!

**Pull request merged**